cmd/init.go

```diff
@@ -90,86 +90,103 @@ to quickly create a Cobra application.`,
-		Trackers[trackerStage1].Tracker.Increment(int64(1))
-		//trackProgress(1, false)
+		trackers[pkg.TrackerStage0].Tracker.Increment(1)
+		trackers[pkg.TrackerStage1].Tracker.Increment(1)
+
 		// todo: this doesn't default to testing the dns check
 		skipHostedZoneCheck := viper.GetBool("init.hostedzonecheck.enabled")
 		if !skipHostedZoneCheck {
 			log.Println("skipping hosted zone check")
 		} else {
-			testHostedZoneLiveness(hostedZoneName, hostedZoneId)
+			aws.TestHostedZoneLiveness(hostedZoneName, hostedZoneId)
 		}
-		Trackers[trackerStage2].Tracker.Increment(int64(1))
+		trackers[pkg.TrackerStage2].Tracker.Increment(1)
 
 		log.Println("calling createSshKeyPair() ")
 		createSshKeyPair()
 		log.Println("createSshKeyPair() complete")
-		Trackers[trackerStage3].Tracker.Increment(int64(1))
+		trackers[pkg.TrackerStage3].Tracker.Increment(1)
 
 		log.Println("calling cloneGitOpsRepo()")
 		cloneGitOpsRepo()
 		log.Println("cloneGitOpsRepo() complete")
-		Trackers[trackerStage4].Tracker.Increment(int64(1))
+		trackers[pkg.TrackerStage4].Tracker.Increment(1)
 
 		log.Println("calling download()")
-		download()
+		err = downloadManager.DownloadTools(config, trackers)
+		if err != nil {
+			panic(err)
+		}
+
 		log.Println("download() complete")
 
-		log.Println("calling getAccountInfo()")
-		getAccountInfo()
-		log.Println("getAccountInfo() complete")
-		Trackers[trackerStage6].Tracker.Increment(int64(1))
+		log.Println("calling GetAccountInfo()")
+		aws.GetAccountInfo()
+		log.Println("GetAccountInfo() complete")
+		trackers[pkg.TrackerStage6].Tracker.Increment(1)
 
-		log.Println("calling bucketRand()")
-		bucketRand()
-		log.Println("bucketRand() complete")
+		log.Println("calling BucketRand()")
+		aws.BucketRand()
+		log.Println("BucketRand() complete")
 
 		log.Println("calling detokenize()")
-		detokenize(fmt.Sprintf("%s/.kubefirst/gitops", home))
+		detokenize(fmt.Sprintf("%s/.kubefirst/gitops", config.HomePath))
 		log.Println("detokenize() complete")
-		Trackers[trackerStage8].Tracker.Increment(int64(1))
+		trackers[pkg.TrackerStage8].Tracker.Increment(1)
 
 		// modConfigYaml()
 		metricName = "kubefirst.init.completed"
 
-		if !dryrunMode {
+		if !config.DryRun {
 			telemetry.SendTelemetry(metricDomain, metricName)
 		} else {
 			log.Printf("[#99] Dry-run mode, telemetry skipped:  %s", metricName)
 		}
 
 		viper.WriteConfig()
-		Trackers[trackerStage9].Tracker.Increment(int64(1))
+		trackers[pkg.TrackerStage9].Tracker.Increment(1)
 		time.Sleep(time.Millisecond * 100)
 	},
 }
 
 func init() {
+	config := configs.ReadConfig()
 	rootCmd.AddCommand(initCmd)
 
-	initCmd.Flags().String("hosted-zone-name", "", "the domain to provision the kubefirst platofrm in")
-	initCmd.MarkFlagRequired("hosted-zone-name")
+	initCmd.Flags().String("hosted-zone-name", "", "the domain to provision the kubefirst platform in")
+	err := initCmd.MarkFlagRequired("hosted-zone-name")
+	if err != nil {
+		panic(err)
```

Comment:
I agree, Im doing a quick refactor to avoid merging conflicts on the next/new PRs, and there is still some heavy work to be done. I'll focus on the re-structure, than we can keep improving.

In this cases, Im just adding the error handling because the error wasn't been treated before.